content/blog/2019-09-25-exceptions-in-bril.md

```diff
@@ -0,0 +1,155 @@
++++
+title = "Exceptions in Bril"
+extra.author = "Rolph Recto"
+extra.bio = """
+[Rolph Recto](rolph-recto.github.io) is a third-year graduate student studying
+the intersection of programming languages, security, and distributed systems.
+"""
++++
+
+
+## Overview
+
+We add exceptions as a control structure in Bril to support non-local
+control flow. 
+This allows frontends to compile exceptions directly.
+The ultimate goal is to extend Bril further with more powerful control
+structures such as first-class continuations or algebraic effects.
+
+
+## Function Calls
+
+To make the implementation of exceptions nontrivial and to make exceptions
+actually be non-local control structures, we first extend Bril so that it
+supports function calls.
+We do this by adding a `call` instruction that takes as arguments a
+function name and a list of function arguments.
+We also extended function declarations to allow for formal parameters.
+The Bril text format now allows for declarations of the form
+
+```
+funcName arg1:type1 ... argN:typeN { instrs }
+```
+
+Our implemention explicitly represents the program stack as an object. 
+
+The interpreter has four configuration variables that consist of its current
```

Comment:
```suggestion
The interpreter has four configuration variables that make up its current
```

That is, the AR consists of four parts, rather than the other way around?

Reply:
Also, I might suggest not saying "configuration variables"—I know what you're going for here (a machine "configuration") but it sounds like the interpreter has a config file.

Reply:
Thanks, that does sound a little stilted!